OneCommunityGlobal/HighestGoodNetworkApp
# Olga_projectSummaries_design_and_connect_to_BE

# Description
Added individual project summary design, connected bmdashboard to backend and set up Redux store for bmprojects

## Main changes explained:
-Added error state to BMDashboard to display error in case of client or server side errors
-Added project actions and reducer to store bmprojects data in Redux store
-Added ErrorAlert component to be displayed when no projects are selected
-Connected to the backend to get BMproject data from the database
-Added design and data to project summary component to display vital information about each project

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. for backend check into development branch
4. `npm run build` and `npm run start` to start the server
6. navigate to http://localhost:3000/bmdashboard
7. verify that:
   - summaries are displayed on the page
   - the page and design are responsive to different screen sizes
   -  error alert is displayed when 'Go to project dashboard' button is clicked without selecting a project
8. Test error handling:
  - To test error handling on initial page load, you can deactivate the backend server and reload the page. This will load an error page with code 503.
  - You can also change the fetch route in src/actions/bmdashboard/projectActions.js > fetchAllProjects from axios.get(ENDPOINTS.BM_PROJECTS_LIST) to axios.get(ENDPOINTS.BM_PROJECTS_LIST + 's'), then reload the page, to force a 404 error.
  - In actions/bmdashboard/projectActions change setProjects(res.data) to setProjects([]), this should result in a message 'No project data' displayed on the screen.

## Photo of added design:
Laptop view:

![Screen Shot 2023-12-05 at 11 57 38 AM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/ecd25431-a358-47ff-b360-55374adee7ad)

Mobile view:

![Screen Shot 2023-12-05 at 11 58 16 AM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/cc76f272-a7b9-4610-a588-038e5d4b2879)

## Note:
The data for project summaries is at the moment simplistic and will likely be replaced later when the project is more developed. Some data is hardcoded on the frontend because those APIs haven't been built out yet.
